airflow/gcp/hooks/campaign_manager.py

```diff
@@ -0,0 +1,244 @@
+        """
+        Creates a report.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report: The report resource to be inserted.
+        :type report: Dict[str, Any]
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .insert(profileId=profile_id, body=report)
+            .execute(num_retries=self.num_retries)
+        )
+        return response
+
+    def list(
+        self,
+        profile_id: str,
+        max_results: Optional[int] = None,
+        scope: Optional[str] = None,
+        sort_field: Optional[str] = None,
+        sort_order: Optional[str] = None,
+    ) -> list:
+        """
+        Retrieves list of reports.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param max_results: Maximum number of results to return.
+        :type max_results: Optional[int]
+        :param scope: The scope that defines which results are returned.
+        :type scope: Optional[str]
+        :param sort_field: The field by which to sort the list.
+        :type sort_field: Optional[str]
+        :param sort_order: Order of sorted results.
+        :type sort_order: Optional[str]
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .list(
+                profileId=profile_id,
+                maxResults=max_results,
+                scope=scope,
+                sortField=sort_field,
+                sortOrder=sort_order,
+            )
+            .execute(num_retries=self.num_retries)
+        )
+        reports = [r for r in response.get('items', [])]
+        next_token = response.get('nextPageToken')
+        while next_token:
+            response = (
+                self.get_conn()  # pylint: disable=no-member
+                    .reports()
+                    .list(
+                    profileId=profile_id,
+                    maxResults=max_results,
+                    pageToken=next_token,
+                    scope=scope,
+                    sortField=sort_field,
+                    sortOrder=sort_order,
+                )
+                .execute(num_retries=self.num_retries)
+            )
+            next_token = response.get('nextPageToken')
+            reports += [r for r in response.get('items', [])]
+
+        return reports
+
+    def patch(self, profile_id: str, report_id: str) -> Any:
+        """
+        Updates a report. This method supports patch semantics.
+
+        :param profile_id: The DFA user profile ID.
+        :type profile_id: str
+        :param report_id: The ID of the report.
+        :type report_id: str
+        """
+        response = (
+            self.get_conn()  # pylint: disable=no-member
+            .reports()
+            .patch(profileId=profile_id, reportId=report_id)
```

Comment:
Is it works? I think, the body parameter is missed.

Reply:
Good catch!